gdx/src/com/badlogic/gdx/maps/tiled/BaseTmjMapLoader.java

```diff
@@ -690,98 +693,94 @@ else if (compression.equals("zlib"))
+		int margin = element.getInt("margin", 0);
 
-			// Tiles
-			JsonValue tiles = element.get("tiles");
+		int offsetX = 0, offsetY = 0;
+		JsonValue offset = element.get("tileoffset");
+		if (offset != null) {
+			offsetX = offset.getInt("x", 0);
+			offsetY = offset.getInt("y", 0);
+		}
 
-			if (tiles == null) {
-				tiles = new JsonValue(JsonValue.ValueType.array);
-			}
+		FileHandle image = resolveTilesetImage(element, tmjFile);
+		String imageSource = element.getString("image", null);
+		int imageWidth = element.getInt("imagewidth", 0);
+		int imageHeight = element.getInt("imageheight", 0);
 
-			addStaticTiles(tmjFile, imageResolver, tileSet, element, tiles, name, firstgid, tilewidth, tileheight, spacing, margin,
-				source, offsetX, offsetY, imageSource, imageWidth, imageHeight, image);
+		TiledMapTileSet tileSet = new TiledMapTileSet();
+		tileSet.setName(name);
+		final MapProperties tileSetProperties = tileSet.getProperties();
+		tileSetProperties.put("firstgid", firstgid);
 
-			Array<AnimatedTiledMapTile> animatedTiles = new Array<>();
+		JsonValue properties = element.get("properties");
+		if (properties != null) {
+			loadProperties(tileSetProperties, properties);
+		}
 
-			for (JsonValue tileElement : tiles) {
-				int localtid = tileElement.getInt("id", 0);
-				TiledMapTile tile = tileSet.getTile(firstgid + localtid);
-				if (tile != null) {
-					AnimatedTiledMapTile animatedTile = createAnimatedTile(tileSet, tile, tileElement, firstgid);
-					if (animatedTile != null) {
-						animatedTiles.add(animatedTile);
-						tile = animatedTile;
-					}
-					addTileProperties(tile, tileElement);
-					addTileObjectGroup(tile, tileElement);
+		JsonValue tiles = element.get("tiles");
+		if (tiles == null) tiles = new JsonValue(JsonValue.ValueType.array);
+
+		addStaticTiles(tmjFile, imageResolver, tileSet, element, tiles, name, firstgid, tilewidth, tileheight, spacing, margin,
+			offsetX, offsetY, imageSource, imageWidth, imageHeight, image);
+
+		Array<AnimatedTiledMapTile> animatedTiles = new Array<>();
+		for (JsonValue tileElement : tiles) {
+			int localtid = tileElement.getInt("id", 0);
+			TiledMapTile tile = tileSet.getTile(firstgid + localtid);
+			if (tile != null) {
+				AnimatedTiledMapTile animatedTile = createAnimatedTile(tileSet, tile, tileElement, firstgid);
+				if (animatedTile != null) {
+					animatedTiles.add(animatedTile);
+					tile = animatedTile;
 				}
+				addTileProperties(tile, tileElement);
+				addTileObjectGroup(tile, tileElement);
 			}
-			// replace original static tiles by animated tiles
-			for (AnimatedTiledMapTile animatedTile : animatedTiles) {
-				tileSet.putTile(animatedTile.getId(), animatedTile);
-			}
+		}
+		for (AnimatedTiledMapTile animatedTile : animatedTiles) {
+			tileSet.putTile(animatedTile.getId(), animatedTile);
+		}
 
-			map.getTileSets().addTileSet(tileSet);
+		return tileSet;
+	}
 
+	private JsonValue resolveTilesetElement (JsonValue element, FileHandle tmjFile) {
```

Comment:
I feel like with almost everything else here `protected` or `public`, adding new `private` methods is something to be careful with. If they will never need customization, or if users can be expected to write their own resolution methods, this is probably fine, but I think `protected` would likely be better for future usage in unforeseen circumstances. Not a dealbreaker.